test/erc20_gateway/erc20_gateway/deposit.js

```diff
@@ -0,0 +1,145 @@
+    };
+
+    setupParam = {
+      metachainId: Utils.getRandomHash(),
+      erc20Cogateway,
+      stateRootProvider,
+      maxStorageRootItems: new BN(50),
+      coGatewayOutboxIndex: new BN(1),
+    };
+
+    await erc20Gateway.setup(
+      setupParam.metachainId,
+      setupParam.erc20Cogateway,
+      setupParam.stateRootProvider,
+      setupParam.maxStorageRootItems,
+      setupParam.coGatewayOutboxIndex,
+    );
+  });
+
+  contract('Positive Tests', async () => {
+    it('should successfully deposit the value token', async () => {
+      await valueToken.approve(
+        erc20Gateway.address,
+        param.amount,
+        { from: depositor },
+      );
+
+      const depositorBalanceBeforeTransfer = await valueToken.balanceOf(depositor);
+      const erc20ContractBalanceBeforeTransfer = await valueToken.balanceOf(erc20Gateway.address);
+
+      const actualMessageHash = await erc20Gateway.deposit.call(
+        param.valueToken,
+        param.amount,
+        param.beneficiary,
+        param.feeGasPrice,
+        param.feeGasLimit,
+        { from: depositor },
+      );
+
+      const channelIdentifier = await erc20Gateway.outboundChannelIdentifier.call();
+
+      const depositIntentHash = HashDepositIntent.getDepositIntentHash(
+        param.valueToken,
+        param.amount,
+        param.beneficiary,
+      );
+
+      const calculatedMessageHash = MessageBusUtils.hashMessage(
+        depositIntentHash,
+        new BN(0),
+        param.feeGasPrice,
+        param.feeGasLimit,
+        depositor,
+        channelIdentifier,
+      );
+
+      await erc20Gateway.deposit(
+        param.valueToken,
+        param.amount,
+        param.beneficiary,
+        param.feeGasPrice,
+        param.feeGasLimit,
+        { from: depositor },
+      );
+
+      const depositorBalanceAfterTransfer = await valueToken.balanceOf(depositor);
+      const erc20ContractBalanceAfterTransfer = await valueToken.balanceOf(erc20Gateway.address);
+
+      assert.isOk(
+        depositorBalanceAfterTransfer.eq(depositorBalanceBeforeTransfer.sub(param.amount)),
+        'Deposit amount must be transferred from depositor.'
+      + ` Expected balance is ${depositorBalanceBeforeTransfer.sub(param.amount).toString(10)} but`
+      + `found ${depositorBalanceAfterTransfer.toString(10)}`,
+      );
+
+      assert.isOk(
+        erc20ContractBalanceAfterTransfer.eq(erc20ContractBalanceBeforeTransfer.add(param.amount)),
+        'Deposit amount must be transferred to ERC20Gateway.'
+      + ` Expected balance is ${erc20ContractBalanceBeforeTransfer.add(param.amount).toString(10)} but`
+      + `found ${depositorBalanceAfterTransfer.toString(10)}`,
+      );
+
+      assert.strictEqual(
+        actualMessageHash,
+        calculatedMessageHash,
+        'Actual message hash must match with calculated message hash.'
+      + `Expected message hash is ${calculatedMessageHash}`
+      + `found ${actualMessageHash}`,
+      );
```

Comment:
Also add test for `outbox[messageHash_] = true;` which is set in the contract.

Reply:
Done.